manager/controllers/systems_advisories_view.go

```diff
@@ -33,32 +34,84 @@ type systemsAdvisoriesDBLoad struct {
 
 var systemsAdvisoriesSelect = database.MustGetSelect(&systemsAdvisoriesDBLoad{})
 
-func systemsAdvisoriesQuery(acc int, systems []SystemID, advisories []AdvisoryName) *gorm.DB {
-	query := database.SystemAdvisories(database.Db, acc).
-		Select(systemsAdvisoriesSelect).
-		Joins("join advisory_metadata am on am.id = sa.advisory_id").
-		Order("sp.inventory_id, am.id")
+func systemsAdvisoriesQuery(acc int, systems []SystemID, advisories []AdvisoryName,
+	limit, offset *int) (*gorm.DB, error) {
+	sysq := database.Systems(database.Db, acc).
+		Distinct("sp.id, sp.inventory_id").
+		// we need to join system_advisories to make `limit` work properly
+		// without this join it can happen that we display less items on some pages
+		Joins("JOIN system_advisories sa ON sa.system_id = sp.id AND sa.rh_account_id = ?", acc).
+		Where("when_patched IS NULL").
+		Order("sp.inventory_id")
 	if len(systems) > 0 {
-		query = query.Where("sp.inventory_id::text in (?)", systems)
+		sysq = sysq.Where("sp.inventory_id::text in (?)", systems)
+	}
+
+	err := Paginate(sysq, limit, offset)
+	if err != nil {
+		return nil, err
 	}
+	query := database.Db.Table("(?) as sp", sysq).
+		Select(systemsAdvisoriesSelect).
+		Joins("JOIN system_advisories sa ON sa.system_id = sp.id AND sa.rh_account_id = ?", acc).
```

Comment:
Is it necessary to join `system_advisories` here when they are alredy joined in `sysq` with the same conditions?

Reply:
`sysq` selects only inventory_id ad in this query I need to join with `sa.advisory_id`. If I selected also `sa.advisory_id` in `sysq`, then the result would have more rows than the amount of systems and therefore the limit would not work properly. That's the reason why I am joining `system_advisories` twice. Do you have a better idea how to do it?